### PR TITLE
pytest integration tests

### PR DESCRIPTION
**Pull request by parente**
* Port server liveliness test from make to test/test_notebook.py
* Add placeholders for base-notebook tests
* Update travis config to install test dependencies on the host

### CODE REVIEW

**Comment by parente** [comment]
Probably good enough for review now. It doesn't add tests of specific kernels and containers yet, but it's a reasonable starting point I think.

File: base-notebook/test/test_container_options.py [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+    c = container.run(
+        user='1010:1010',
+        group_add=['users'],
+        command=['start.sh', 'id']
+    )
+    rv = c.wait(timeout=5)
+    assert rv == 0
+    assert 'uid=1010 gid=1010 groups=1010,100(users)' in c.logs(stdout=True).decode('utf-8')
+
+@pytest.mark.dev
```
Comment (parente) [inline base-notebook/test/test_container_options.py]: Need to remove this.

Comment (parente) [reply base-notebook/test/test_container_options.py]: And from the test above.

File: base-notebook/test/test_container_options.py [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+    echo "test content" > test.txt
+    cat test.txt
+    ''')
+    script.chmod(0o755)
+
+    c = container.run(
+        volumes={
+            home.strpath: {'bind': '/home/jovyan', 'mode': 'rw'},
+        },
+        command=['start.sh', '/home/jovyan/test.sh']
```
Comment (parente) [inline base-notebook/test/test_container_options.py]: This test script should change to trying to write to a subdirectory that is not chown'ed to `jovyan:users`on the host and then to one that is. It should then assert the first fails and the second succeeds. Right now, it's not testing much

File: .travis.yml [hunk 3]
```diff
@@ -1,8 +1,10 @@
...
-
+language: python
+python:
+    - 3.6
 sudo: required
 services:
     - docker
-
+install:
+    - pip install docker pytest requests
```
Comment (ericdill) [inline .travis.yml]: should this be moved in to the `make test` command? Otherwise it seems to make it slightly more challenging to run the tests

**Comment by parente** [comment]
Latest test failing because of a jupyterlab / jupyter-widgets version conflict. Updating jupyterlab and to 0.30.* and trying again.

**Comment by parente** [comment]
Merging to get the JLab compatibility fix in. Looks like master rebuilds are broken at the moment because of it.

**Comment by minrk** [comment]
Hooray! Thanks, @parente!

**Pull request merged**